File: NEC_Challenge.Core/Models/CryptoCurrency.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NEC_Challenge.Core.Models
{
    public class CryptoCurrency
    {
        public long Id { get; set; }
        public string Symbol { get; set; }
        public string Name { get; set; }
        public double PriceInUsd { get; set; }
        public int AmountToConvert { get; set; }
    }
}

File: NEC_Challenge.Web/src/app/core/models/crypto-currency.ts
export class CryptoCurrency {
    id: number;
    symbol: string;
    name: string;
    priceInUsd: number;
    amountToConvert: number;

    constructor(_id: number, _symbol: string, _name: string, _priceInUsd: number, _amountToConvert: number) 
        {
        this.id = _id;
        this.symbol = _symbol;
        this.name = _name;
        this.priceInUsd = _priceInUsd;
        this.amountToConvert = _amountToConvert;
    }

    public static adapt(item: any): CryptoCurrency {
        return new CryptoCurrency(
            item.id,
            item.symbol,
            item.name,
            item.priceInUsd,
            item.amountToConvert
        );
    }
}